tutorials/scripting/gdscript/gdscript_basics.rst

```diff
@@ -1454,6 +1454,31 @@ Here are some examples of expressions::
 Identifiers, attributes, and subscripts are valid assignment targets. Other expressions cannot be on the left side of
 an assignment.
 
+self
+^^^^
+
+``self`` can be used to refer to the current instance and is often equivalent to directly referring to symbols available in
+the current script, however, it also allows you to access properties, methods, and other names in subclasses of the current script.
+
+::
+
+    class_name Weapon extends Node
+
+    func _ready() -> void:
+        print(self.damage)
+
+``damage`` is not defined in ``Weapon``, but we (through, e.g., :ref:`documenting our code <doc_gdscript_documentation_comments>`)
+can assume that it will exist in subclasses of ``Weapon``. For example:
```

Comment:
I think mentioning doc comments is unnecessary here. You cannot document non-existent members, except by mentioning them in the class description.